ui/app/AppLayouts/Chat/popups/RequestPaymentModal.qml

```diff
@@ -0,0 +1,228 @@
+import QtQuick 2.15
+import QtQuick.Layouts 1.15
+import QtQml.Models 2.15
+
+import StatusQ 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components.private 0.1 as SQP
+import StatusQ.Components 0.1
+import StatusQ.Popups.Dialog 0.1
+
+import AppLayouts.Wallet.controls 1.0
+import AppLayouts.Wallet.adaptors 1.0
+
+import shared.popups.send.views 1.0
+import shared.controls 1.0
+import shared.stores 1.0 as SharedStores
+import utils 1.0
+
+StatusDialog {
+    id: root
+
+    required property SharedStores.RequestPaymentStore store
+
+    property int selectedNetworkChainId: Constants.chains.mainnetChainId
+    property string selectedAccountAddress
+    property string selectedTokenKey: Constants.ethToken
+
+    readonly property string amount: {
+        if (!d.isSelectedHoldingValidAsset || !d.selectedHolding.item.marketDetails || !d.selectedHolding.item.marketDetails.currencyPrice) {
+            return "0"
+        }
+        return amountToSendInput.text
+    }
+
+    objectName: "requestPaymentModal"
+
+    implicitWidth: 480
+    implicitHeight: 470
+
+    modal: true
+    padding: 0
+    backgroundColor: Theme.palette.statusModal.backgroundColor
+
+    title: qsTr("Payment request")
+
+    onOpened: {
+        // Setting value here because to prevent not updating when selected token key is filled
+        d.selectedHolding.value = Qt.binding(function() { return root.selectedTokenKey })
+
+        if (!!root.selectedTokenKey) {
+            holdingSelector.setSelection(d.selectedHolding.item.symbol, d.selectedHolding.item.iconSource, d.selectedHolding.item.tokensKey)
+        }
+    }
+
+    QtObject {
+        id: d
+
+        readonly property ModelEntry selectedHolding: ModelEntry {
+            sourceModel: holdingSelector.model
+            key: "tokensKey"
+        }
+
+        readonly property bool isSelectedHoldingValidAsset: !!selectedHolding.item
+
+        readonly property var adaptor: TokenSelectorViewAdaptor {
```

Comment:
We have this warning in storybook:
`file:///status-desktop/ui/app/AppLayouts/Wallet/adaptors/TokenSelectorViewAdaptor.qml:285:17: Unable to assign [undefined] to QAbstractItemModel*`